### PR TITLE
CORE-664 | Fix workflows & python version

### PR DESCRIPTION
**Pull request by Lexicality**
## Jira Ticket
CORE-664

## Description
Since everything's installing this thing

## Author Checklist

-   [x] Made things better
-   [ ] Tests cover all the edge cases
-   [x] Not left any stray logging or comments
-   [x] Useful commit messages
-   [x] Bumped version in pyproject.toml

## Reviewer Checklist

-   [ ] I've read everything and understand it
-   [ ] I've followed our [review guidelines](https://lantum.atlassian.net/wiki/spaces/ENG/pages/29982738/Code+Reviews)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.6.2,<3.10"
+python = ">=3.7,<3.13"
```
Comment (marekzp) [inline pyproject.toml:12]: woot woot!

**Review by marekzp — APPROVED** [review]
